exporters/geneva/src/exporter.cc

```diff
@@ -78,50 +78,85 @@ opentelemetry::sdk::common::ExportResult Exporter::Export(
               metric_data.end_ts, metric_data.instrument_descriptor.name_,
               point_data_with_attributes.attributes);
           data_transport_->Send(event_type, buffer_non_histogram_,
                                 body_length + kBinaryHeaderSize);
 
-        } else if (nostd::holds_alternative<sdk::metrics::LastValuePointData>(
-                       point_data_with_attributes.point_data)) {
+        } 
+        else if (nostd::holds_alternative<sdk::metrics::LastValuePointData>(
+                       point_data_with_attributes.point_data)) 
+        {
           auto value = nostd::get<sdk::metrics::SumPointData>(
               point_data_with_attributes.point_data);
-          MetricsEventType event_type = MetricsEventType::ULongMetric;
-          if (nostd::holds_alternative<double>(value.value_)) {
-            event_type = MetricsEventType::DoubleMetric;
+          ValueType new_value = value.value_;
+          if (nostd::holds_alternative<int64_t>(value.value_)) 
+          {
+              // NOTE - Potential for minor precision loss implicitly going from int64_t to double -
+              //   - A 64-bit integer can hold more significant decimal digits than a standard 
+              //     IEEE (64-bit) double precision floating-point representation
+              new_value = static_cast<double>(nostd::get<int64_t>(new_value));
           }
+          MetricsEventType event_type = MetricsEventType::DoubleMetric;
           body_length = SerializeNonHistogramMetrics(
               sdk::metrics::AggregationType::kLastValue, event_type,
-              value.value_, metric_data.end_ts,
+              new_value, metric_data.end_ts,
               metric_data.instrument_descriptor.name_,
               point_data_with_attributes.attributes);
           data_transport_->Send(event_type, buffer_non_histogram_,
                                 body_length + kBinaryHeaderSize);
-        } else if (nostd::holds_alternative<sdk::metrics::HistogramPointData>(
-                       point_data_with_attributes.point_data)) {
+        } 
+        else if (nostd::holds_alternative<sdk::metrics::HistogramPointData>(
+                       point_data_with_attributes.point_data)) 
+        {
           auto value = nostd::get<sdk::metrics::HistogramPointData>(
               point_data_with_attributes.point_data);
-          MetricsEventType event_type =
-              MetricsEventType::ExternallyAggregatedULongDistributionMetric;
-          if (nostd::holds_alternative<double>(value.sum_)) {
-            LOG_ERROR("Geneva Exporter::Export- Double value is not supported "
-                      "for Histogram");
-            continue;
+          ValueType new_sum = value.sum_;
+          ValueType new_min = value.min_;
+          ValueType new_max = value.max_;
+
+          if (nostd::holds_alternative<double>(value.sum_)) 
+          {
+            // Double is not supported by Geneva, convert it to int64_t
```

Comment:
Will it be supported later? If so, probably add a TODO?

Reply:
Yes, we have to modify our code once Geneva start supporting double. Have added TODO for now.